src/subscribe.rs

```diff
@@ -0,0 +1,429 @@
+#[ft_sdk::form]
+fn subscribe(
+    ft_sdk::Query(name): ft_sdk::Query<"name", Option<String>>,
+    ft_sdk::Query(phone): ft_sdk::Query<"phone", Option<String>>,
+    ft_sdk::Query(email): ft_sdk::Query<"email", Option<String>>,
+    ft_sdk::Query(source): ft_sdk::Query<"source", Option<String>>,
+    ft_sdk::Query(topic): ft_sdk::Query<"topic", Option<String>>,
+    ft_sdk::Query(next): ft_sdk::Query<"next", Option<String>>,
+    sid: ft_sdk::Cookie<{ ft_sdk::auth::SESSION_KEY }>,
+    host: ft_sdk::Host,
+    mountpoint: ft_sdk::Mountpoint,
+    mut conn: ft_sdk::Connection,
+) -> ft_sdk::form::Result {
+    use diesel::prelude::*;
+
+    let (subscriber, id) = validate(&mut conn, name, phone, email, sid)?;
+    let user_id = get_or_create_user_id(&mut conn, id, &subscriber)?;
+
+    #[derive(diesel::QueryableByName)]
+    #[diesel(table_name = ft_sdk::auth::fastn_user)]
+    struct UserData {
+        data: String,
+    }
+
+    ft_sdk::println!("querying data");
+    let data: serde_json::Value = match diesel::sql_query(
+        r#"
+        SELECT data from fastn_user where id = $1 LIMIT 1;
+        "#,
+    )
+    .bind::<diesel::sql_types::Integer, _>(user_id.0)
+    .get_result::<UserData>(&mut conn)
+    {
+        Ok(d) => serde_json::from_str(&d.data)?,
+        Err(e) => return Err(e.into()),
+    };
+
+    ft_sdk::println!("querying data 2");
+
+    let (subscribed, mut data) = add_subscription_info(data, topic.clone(), source, &subscriber);
+
+    if subscribed {
+        let name = subscriber.name.unwrap_or_else(|| "".to_string());
+        let key = ft_sdk::Rng::generate_key(64);
+        data = add_confirmation_key_in_user(data, &key);
+
+        let conf_link = confirmation_link(&key, &subscriber.email, &host, &mountpoint);
+        send_double_opt_in_email(&mut conn, (&name, &subscriber.email), &conf_link, topic)?;
+    }
+
+    let data = serde_json::to_string(&data)?;
+    ft_sdk::println!("querying data 3");
+
+    diesel::update(
+        ft_sdk::auth::fastn_user::table.filter(ft_sdk::auth::fastn_user::id.eq(user_id.0)),
+    )
+    .set((
+        ft_sdk::auth::fastn_user::data.eq(&data),
+        ft_sdk::auth::fastn_user::updated_at.eq(ft_sdk::env::now()),
+    ))
+    .execute(&mut conn)?;
```

Comment:
Fix this to use `ft_sdk::auth::provider::update_user`

Reply:
I have explained why this can't happen today, read my replies above.